src/Marten/Events/Daemon/Coordination/ProjectionCoordinator.cs

```diff
@@ -0,0 +1,252 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using JasperFx.Core;
+using Marten.Events.Daemon.Resiliency;
+using Marten.Storage;
+using Microsoft.Extensions.Logging;
+using Polly;
+
+namespace Marten.Events.Daemon.Coordination;
+
+public class ProjectionCoordinator<T>: ProjectionCoordinator, IProjectionCoordinator<T> where T : IDocumentStore
+{
+    public ProjectionCoordinator(T documentStore, ILogger<ProjectionCoordinator> logger) : base(documentStore, logger)
+    {
+    }
+}
+
+public class ProjectionCoordinator : IProjectionCoordinator
+{
+    private readonly StoreOptions _options;
+    private readonly ILogger<ProjectionCoordinator> _logger;
+
+    private readonly ConcurrentDictionary<IMartenDatabase, IProjectionDaemon> _daemons = new();
+    private readonly ResiliencePipeline _resilience;
+    private readonly CancellationTokenSource _cancellation = new();
+    private Task _runner;
+
+    public ProjectionCoordinator(IDocumentStore documentStore, ILogger<ProjectionCoordinator> logger)
+    {
+        var store = (DocumentStore)documentStore;
+
+        if (store.Options.Projections.AsyncMode == DaemonMode.Solo)
+        {
+            Distributor = new SoloProjectionDistributor(store);
+        }
+        else if (store.Options.Projections.AsyncMode == DaemonMode.HotCold)
+        {
+            if (store.Options.Tenancy is DefaultTenancy)
+            {
+                Distributor = new SingleTenantProjectionDistributor(store);
+            }
+            else
+            {
+                Distributor = new MultiTenantedProjectionDistributor(store);
+            }
+        }
+
+        _options = store.Options;
+        _logger = logger;
+        _resilience = store.Options.ResiliencePipeline;
+
+        Store = store;
+    }
+
+    public IProjectionDaemon DaemonForMainDatabase()
+    {
+        var database = (MartenDatabase)Store.Tenancy.Default.Database;
+        if (_daemons.TryGetValue(database, out var daemon))
+        {
+            return daemon;
+        }
+
+        daemon = database.StartProjectionDaemon(Store, _logger);
+        _daemons[database] = daemon;
+
+        return daemon;
+    }
+
+    public async ValueTask<IProjectionDaemon> DaemonForDatabase(string databaseIdentifier)
```

Comment:
**QUESTION:** I don't see that used anywhere. Is it a future concept?

Also minor, but if it's needed, then it may be worth adding private method  `ValueTask<IProjectionDaemon> DaemonForDatabase(MartenDatabase database)` unifying the handling in both `DaemonForMainDatabase` methods.

Reply:
Everything around IProjectionCoordinator is "future use" pending whatever ends up happening here: https://github.com/JasperFx/marten/issues/2952. The mechanics for accessing the main database and a named database isn't actually quite the same here, so I wouldn't worry about unifying them.

Reply:
Nope, you were right. Good suggestion. New code will have this:

```csharp
    private IProjectionDaemon findDaemonForDatabase(MartenDatabase database)
    {
        if (_daemons.TryFind(database, out var daemon))
        {
            return daemon;
        }

        lock (_daemonLock)
        {
            if (_daemons.TryFind(database, out daemon))
            {
                return daemon;
            }

            daemon = database.StartProjectionDaemon(Store, _logger);
            _daemons = _daemons.AddOrUpdate(database, daemon);
        }

        return daemon;
    }
```